### DISCUSSION TITLE
Consumer offset tracking

### DISCUSSION THREAD

**Discussion by steache** [php-amqplib/php-amqplib#1074] [Q&A] [answered]
Hello,

is it possible to enable automatic offset tracking for consumers via some argument?

https://blog.rabbitmq.com/posts/2021/09/rabbitmq-streams-offset-tracking/

I did not find anything related to this.

I'm trying to use super-streams, and that is also not possible without configuring specific partition for example "{stream}-{partition}" -> "stream-0"

Thank you for any answer. :)

**Comment by lukebakken** [comment, marked as answer]
Hello! Let me ask around to see if anyone on Team RabbitMQ wants to tackle this.

**Reply by steache** [reply]
It looks like it's more client-side thing.

In java it's possible to enable auto-tracking, but it just stores the offset in the stream every X events. 

(https://github.com/rabbitmq/rabbitmq-stream-java-client/blob/b32997637fc41599e829a7cdf40be3ecf784b670/src/main/java/com/rabbitmq/stream/impl/OffsetTrackingCoordinator.java#L206)

I can see implementing myself the offset-part, that looks ok.

Maybe I'm more interested in knowing how to handle consuming of super-streams. :)

Thank you very much!

**Reply by lukebakken** [reply]
https://www.rabbitmq.com/stream-core-plugin-comparison.html

This library (`php-amqplib`) is AMQP only, so super streams aren't supported. As far as I know nobody has implemented a PHP client for the Stream Plugin.
